### PR TITLE
Hide private trait type params and show hidden items with document-private

### PR DESCRIPTION
**Pull request by chrisduerr**
As discussed in #46380, this PR removes the `strip-hidden` pass from `--document-private-items` which allows showing `#[doc(hidden)]` with rustdoc.

The second commit removes the trait implementation from the docs if the trait's parameter is private.

### CODE REVIEW

File: src/test/rustdoc/issue-46380-2.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub trait PublicTrait<T> {}
+
+// @has issue_46380_2/struct.Public.html
+pub struct PublicStruct;
+
+// @!has - '//*[@class="impl"]' 'impl Add<Private> for Public'
```
Comment (QuietMisdreavus) [inline src/test/rustdoc/issue-46380-2.rs]: This probably needs to be `'impl PublicTrait<PrivateStruct> for PublicStruct'`, to actually check the right thing.

Comment (chrisduerr) [reply src/test/rustdoc/issue-46380-2.rs]: Oh yeah totally my bad. I renamed things and then only checked that it works, not if the negative fails.

I'll test both failure and success and then push another commit.

**Comment by QuietMisdreavus** [comment]
Cool! r=me is travis is green.

**Comment by QuietMisdreavus** [comment]
@bors r+

Thanks again!

**Pull request merged**